app/controllers/api/v1/user/users_controller.rb

```diff
@@ -6,4 +6,22 @@ def show
     render json: current_user
   end
 
+  def finance_overview
+    ordergroup = Ordergroup.include_transaction_class_sum.find(current_ordergroup.id)
+
+    render json: {
+      finance_overview: {
+        account_balance: ordergroup.account_balance.to_f,
+        available_funds: ordergroup.get_available_funds.to_f,
+        financial_transaction_class_sums: FinancialTransactionClass.sorted.map { |c|
+          {
+            id: c.id,
+            name: c.display,
+            amount: ordergroup["sum_of_class_#{c.id}"]
+          }
+        }
+      }
+    }
+  end
+
```

Comment:
I assume you've thought about whether to do this with ActiveModelSerializer.

Other question: where will we expose Ordergroup information? We don't really, right now, and it is perhaps related to this.

Reply:
i don't see a benefit of a serializer in this case. i'd refactor, if we decide that there's a better way

do you you mean with _Ordergroup information_? we expose our ordergroup-related `finacial_transactions` already via `/user/financial_transactions`.

Reply:
True. The API doesn't currently return e.g. the name of the user's ordergroup, or whether it has one.
If we were to add that, would there be overlap with `financial_overview`?

One way would be to let `/api/user` also return group memberships, and have `/api/ordergroup` show ordergroup fields. Another way would be to only have `/api/user/ordergroup` with ordergroup fields (and then perhaps `/api/user/workgroups` for workgroup memberships). The user's ordergroup endpoint would either include the financial overview, or there would be a separate `/api/user/financial_overview` endpoint.

Since any choice we make will stick for quite a while, I'd like to put some thought to it.
What do you think?

Reply:
IIRK the basic idea of `/api/user` was to make dealing with access roles easier: every user can request `/api/user/financial_transactions` to get her own transactions, but only finance-people can request `/api/financial_transactions` to get **all** transactions.
sticking to that idea `/api/user/finance_overview` should be the way to go. I see no benefit in moving it to `/api/user/ordergroup/finance_overview`, since a user has only one and we would need to move the `/api/user/financial_transactions` to `/api/user/ordergroup/financial_transactions` then too. IMHO renaming `/api/user` to `/api/me` or `/api/my` would make the basic idea more clear, but we settled with `/api/user` already.

Reply:
And what you've done now seems even better, on second thought: putting it in an ordergroup controller. :+1: